src/core/routing/route-diversity.ts

```
@@ -0,0 +1,231 @@
+} from '../utils/route-comparison';
+import { TransitGraph } from '../graph/graph';
+import { getLinePathSignature } from '../utils/route-signature';
+
+/**
+ * Optimize a set of routes to ensure diversity
+ * @returns A subset of routes that are sufficiently diverse
+ */
+export function optimizeRouteDiversity(
+  routes: Route[],
+  graph: TransitGraph,
+  maxRoutes: number = 5
```

Biome (1.9.4)

[error] This type annotation is trivially inferred from its initialization.

Safe fix: Remove the type annotation.


(lint/style/noInferrableTypes)